tests/functional/prometheous/test_prometheus_validations.py

```diff
@@ -974,3 +975,78 @@ def test_restart_prometheus_glusterfs_pod(self):
+        openshift_ops.switch_oc_project(
+            self._master, self._prometheus_project_name)
+        prometheus_pods = openshift_ops.oc_get_pods(
+            self._master, selector=self._prometheus_resources_selector)
+        if not prometheus_pods:
+            self.skipTest(
+                prometheus_pods, "Skipping test as prometheus"
+                " pod is not present")
+
+        # Get one of the prometheus pod name and respective pvc name
+        prometheus_pod = list(prometheus_pods.keys())[0]
+        pvc_custom = ":.spec.volumes[*].persistentVolumeClaim.claimName"
+        pvc_name = openshift_ops.oc_get_custom_resource(
+            self._master, "pod", pvc_custom, prometheus_pod)
+        self.assertTrue(
+            pvc_name, "Failed to get PVC name for prometheus"
+            " pod {}".format(prometheus_pod))
+        pvc_name = pvc_name[0]
+        self.verify_iscsi_sessions_and_multipath(
+            pvc_name, prometheus_pod, rtype='pod',
+            heketi_server_url=self._registry_heketi_server_url,
+            is_registry_gluster=True)
+
+        # Try to fetch metric from the prometheus pod
+        self._fetch_metric_from_promtheus_pod(
+            metric='kube_persistentvolumeclaim_info')
+
+        # Create storage class
+        openshift_ops.switch_oc_project(
+            self._master, self._registry_project)
+        self.sc_name = self.create_storage_class(
+            vol_name_prefix=self.prefix, glusterfs_registry=True)
+        self.addCleanup(openshift_ops.switch_oc_project,
+                        self._master, self._registry_project)
+
+        # Create PVCs and app pods
+        pvc_size, pvc_count, batch_count = 1, 5, 5
+        for _ in range(batch_count):
+            test_pvc_names = self.create_and_wait_for_pvcs(
+                pvc_size, pvc_name_prefix=self.prefix,
+                pvc_amount=pvc_count, sc_name=self.sc_name, timeout=600,
+                wait_step=10)
+            self.create_dcs_with_pvc(
+                test_pvc_names, timeout=600, wait_step=5,
+                dc_name_prefix="autotests-dc-with-app-io",
+                space_to_use=1048576)
+
+        # Check from the prometheus pod for the pvc space usage
+        openshift_ops.switch_oc_project(
+            self._master, self._prometheus_project_name)
+        mount_path = "/prometheus"
+        cmd = "oc exec {0} -- df -PT {1} | grep {1}".format(
+            prometheus_pod, mount_path)
+        out = self.cmd_run(cmd)
+        self.assertTrue(out, "Failed to get info about mounted volume. "
+                             "Output is empty.")
+
+        # Try to fetch metric from prometheus pod
+        self._fetch_metric_from_promtheus_pod(
+            metric='kube_persistentvolumeclaim_info')
+        self._fetch_metric_from_promtheus_pod(
+            metric='kube_pod_spec_volumes_persistentvolumeclaims_info')
+        self.addCleanup(openshift_ops.switch_oc_project,
```

Comment:
Why this cleanup needed here ?

Reply:
If we have the cleanup at line 1050, log snippet after the test case execution is completed:
========================================================================
2021-03-08 13:35:19,324 INFO (tearDown) Ending Test: tests.functional.prometheous.test_metrics_workload_on_prometheus.TestMetricsWorkloadOnPrometheus.test_metrics_workload_on_prometheus : 13_33_08_03_2021
2021-03-08 13:35:19,324 INFO (run) root@10.70.47.18 (cp): oc project glusterfs
2021-03-08 13:35:19,810 INFO (_log_results) ^[[34;1mRETCODE (root@10.70.47.18): 0^[[0m
2021-03-08 13:35:19,811 INFO (_log_results) ^[[47;30;1mSTDOUT (root@10.70.47.18)...
Now using project "glusterfs" on server "https://dhcp47-18.lab.eng.blr.redhat.com:8443".
^[[0m
2021-03-08 13:35:19,811 INFO (run) root@10.70.47.18 (cp): oc scale  --replicas=0 dc/autotests-dc-with-app-io-2qss0lmjndssiw dc/autotests-dc-with-app-io-fdyt7a6ypxvlsu dc/autotests-dc-with-app-io-x92a30rrlxgooq dc/autotests-dc-with-app-io-wafrwrye8pz1hh dc/autotests-dc-with-app-io-lqwpl4ez2toy99
2021-03-08 13:35:20,385 INFO (_log_results) ^[[34;1mRETCODE (root@10.70.47.18): 0^[[0m
2021-03-08 13:35:20,385 INFO (_log_results) ^[[47;30;1mSTDOUT (root@10.70.47.18)...
deploymentconfig.apps.openshift.io/autotests-dc-with-app-io-2qss0lmjndssiw scaled
deploymentconfig.apps.openshift.io/autotests-dc-with-app-io-fdyt7a6ypxvlsu scaled
deploymentconfig.apps.openshift.io/autotests-dc-with-app-io-x92a30rrlxgooq scaled
deploymentconfig.apps.openshift.io/autotests-dc-with-app-io-wafrwrye8pz1hh scaled
deploymentconfig.apps.openshift.io/autotests-dc-with-app-io-lqwpl4ez2toy99 scaled
^[[0m


But if the cleanup is removed, we see following errors:
=============================================
2021-03-17 16:23:29,867 INFO (tearDown) Ending Test: tests.functional.prometheous.test_metrics_workload_on_prometheus.TestMetricsWorkloadOnPrometheus.test_metrics_workload_on_prometheus : 16_21_17_03_2021
2021-03-17 16:23:29,867 INFO (run) root@10.70.47.18 (cp): oc scale  --replicas=0 dc/autotests-dc-with-app-io-bsa6p4yfteoxrb dc/autotests-dc-with-app-io-o63h39lripebbq dc/autotests-dc-with-app-io-3rmbf3byjfkpcz dc/autotests-dc-with-app-io-gum155zb3pzaxe dc/autotests-dc-with-app-io-w7z31tnx9luep2
2021-03-17 16:23:32,433 INFO (_log_results) RETCODE (root@10.70.47.18): 1
2021-03-17 16:23:32,433 INFO (_log_results) STDERR (root@10.70.47.18)...
Error from server (NotFound): deploymentconfigs.apps.openshift.io "autotests-dc-with-app-io-bsa6p4yfteoxrb" not found
Error from server (NotFound): deploymentconfigs.apps.openshift.io "autotests-dc-with-app-io-o63h39lripebbq" not found
Error from server (NotFound): deploymentconfigs.apps.openshift.io "autotests-dc-with-app-io-3rmbf3byjfkpcz" not found
Error from server (NotFound): deploymentconfigs.apps.openshift.io "autotests-dc-with-app-io-gum155zb3pzaxe" not found
Error from server (NotFound): deploymentconfigs.apps.openshift.io "autotests-dc-with-app-io-w7z31tnx9luep2" not found

2021-03-17 16:23:32,434 ERROR (cmd_run) Failed to execute command 'oc scale  --replicas=0 dc/autotests-dc-with-app-io-bsa6p4yfteoxrb dc/autotests-dc-with-app-io-o63h39lripebbq dc/autotests-dc-with-app-io-3rmbf3byjfkpcz dc/autotests-dc-with-app-io-gum155zb3pzaxe dc/autotests-dc-with-app-io-w7z31tnx9luep2' on '10.70.47.18' node. Got non-zero return code '1'. Err: Error from server (NotFound): deploymentconfigs.apps.openshift.io "autotests-dc-with-app-io-bsa6p4yfteoxrb" not found
Error from server (NotFound): deploymentconfigs.apps.openshift.io "autotests-dc-with-app-io-o63h39lripebbq" not found
Error from server (NotFound): deploymentconfigs.apps.openshift.io "autotests-dc-with-app-io-3rmbf3byjfkpcz" not found
Error from server (NotFound): deploymentconfigs.apps.openshift.io "autotests-dc-with-app-io-gum155zb3pzaxe" not found
Error from server (NotFound): deploymentconfigs.apps.openshift.io "autotests-dc-with-app-io-w7z31tnx9luep2" not found